### PR TITLE
fix: remove unwanted space between link enclosed with brackets

### LINKED ISSUE #538: Problem(render): link renders with unwanted spaces
Source: https://docs.google.com/document/d/1_DV8rzW3dGijN0xWSXbV0Iiwd0p7Jrwof9va5R_TWz8/edit
Render: https://holdex.io/c/announcements/about

image

### PR DESCRIPTION
**Pull request by angelicawill**
Time spent: 3 hours

### CODE REVIEW

**Comment by angelicawill** [comment]
The message received by the BodyParser from the google-conversion API has space around the link for links surrounded by brackets, and I still don't understand why.  
CleanShot 2023-12-13 at 15 41 53


So what I do is handling this in the render process, basically just trimming trailing whitespace for texts that ends or starts with brackets. This can be extended for other symbols as well.

This solution solves the problem in the "about" page, but I think this is a band-aid solution and doesn't address the problem at its core, but it's the best I can do now since other tasks are more urgent. 

@zolotokrylin

File: src/lib/components/BodyRenderer/inline.svelte [hunk 1]
```diff
@@ -19,6 +19,15 @@
...
 
+  function trimText(text: string) {
+    if (text.endsWith('( ')) {
+      text = text.trim();
+    }
+    if (text.startsWith(' )')) {
+      text = text.trim();
+    }
+    return text;
+  }
```
Comment (zolotokrylin) [inline src/lib/components/BodyRenderer/inline.svelte:30]: @angelicawill  if I want the case to be like `(\space` it won't work? 
I think it won't work.
Thus, you need to fix this on the level of the `link` render function.

@georgeciubotaru correct me if I am wrong

**Review by zolotokrylin — CHANGES_REQUESTED** [review]

File: src/lib/components/BodyRenderer/inline.svelte (hunk 1 above)
Comment (angelicawill) [reply src/lib/components/BodyRenderer/inline.svelte:30]: @zolotokrylin Sorry, I don't get what you mean. Can you please elaborate more? Thank you

Comment (zolotokrylin) [reply src/lib/components/BodyRenderer/inline.svelte:30]: @angelicawill 
if I have a text in google document like this: 
image

Will it render on our website like this: 
a - `( text )`, or like this:
b - `(text)`
? 

 Should be `a`

Comment (angelicawill) [reply src/lib/components/BodyRenderer/inline.svelte:30]: @zolotokrylin I understand now. That's a case that I failed to recognize

**Pull request merged**